### PR TITLE
Adding evalica

### PR DESCRIPTION
**Pull request by dustalov**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by dustalov** [comment]
@conda-forge/help-python, ready for review!

File: recipes/evalica/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  script:
+    - cargo-bundle-licenses --format yaml --output THIRDPARTY.yml
+    - {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  number: 0
+
+requirements:
+  build:
+    - {{ stdlib("c") }}
+    - {{ compiler('rust') }}
+    - python  # [build_platform != target_platform]
```
Comment (xhochy) [inline recipes/evalica/meta.yaml:24]: If you add this, you also need to add cross-python. Better let the arch migrator do it for you.
```suggestion
```

File: recipes/evalica/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    - cargo-bundle-licenses
+    - maturin
+    - numpy
+    - pandas
```
Comment (xhochy) [inline recipes/evalica/meta.yaml:28]: This should only be in host

File: recipes/evalica/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  host:
+    - python
+    - pip
+    - maturin
+  run:
+    - python
+    - numpy
+    - pandas
+    - libgcc-ng  # [linux]
```
Comment (xhochy) [inline recipes/evalica/meta.yaml:37]: Please add a c compiler instead.
```suggestion
```

File: recipes/evalica/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    - maturin
+    - numpy
+    - pandas
+  host:
+    - python
+    - pip
+    - maturin
+  run:
+    - python
+    - numpy
```
Comment (xhochy) [inline recipes/evalica/meta.yaml:35]: Probably move this to host and let numpy add its run_exports automatically.

**Review by xhochy — CHANGES_REQUESTED** [review]

File: recipes/evalica/meta.yaml (hunk 3 above)
Comment (dustalov) [reply recipes/evalica/meta.yaml]: I followed the example of other recipes that approached this problem this way: https://github.com/search?q=org%3Aconda-forge+libgcc-ng+%5Blinux%5D&type=code.

I tried adding the compiler to `run` as `{{ compiler('c') }}`, but did not seem to work:

```
conda_build.exceptions.OverLinkingError: overlinking check failed 
["  ERROR (evalica,lib/python3.10/site-packages/evalica/evalica.cpython-310-x86_64-linux-gnu.so): .. but ['conda-forge/linux-64::libgcc-ng==14.1.0=h77fa898_0'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)"]
```

File: recipes/evalica/meta.yaml (hunk 4 above)
Comment (dustalov) [reply recipes/evalica/meta.yaml]: The linter requires `python` to be here: https://github.com/conda-forge/staged-recipes/pull/26903#issuecomment-2241075910.

**Comment by dustalov** [comment]
@xhochy, thank you for your comments. I updated `meta.yaml` and addressed most of the comments, but I am not sure about the best way to fix the build errors caused by these changes.

File: recipes/evalica/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    - {{ compiler('rust') }}
+  host:
+    - cargo-bundle-licenses
+    - maturin
+    - numpy
+    - pandas
+    - pip
+    - python
+  run:
+    - {{ compiler('c') }}
```
Comment (xhochy) [inline recipes/evalica/meta.yaml:32]: This should be in `build`, but `python` should stay here.

**Review by xhochy — APPROVED** [review]

File: recipes/evalica/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    - {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('rust') }}
+    - {{ stdlib("c") }}
+  host:
+    - cargo-bundle-licenses
```
Comment (xhochy) [inline recipes/evalica/meta.yaml:26]:
```suggestion
    - cargo-bundle-licenses
  host:
```

**Review by xhochy — CHANGES_REQUESTED** [review]

**Review by xhochy — APPROVED** [review]

**Pull request merged**